teamcity: failed tests on master: testrace/TestSplitSnapshotRace_SplitWins, test/TestSplitSnapshotRace_SplitWins

The following tests appear to have failed:

[#583637](https://teamcity.cockroachdb.com/viewLog.html?buildId=583637):


```
--- FAIL: testrace/TestSplitSnapshotRace_SplitWins (0.000s)
Race detected!

------- Stdout: -------
I180326 15:06:52.432640 105762 storage/store.go:1394  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180326 15:06:52.471270 105762 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46029" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.527940 105762 gossip/gossip.go:1292  [n2] no incoming or outgoing connections
I180326 15:06:52.538151 105883 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:46029
I180326 15:06:52.553567 105762 storage/store.go:1394  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I180326 15:06:52.559421 105762 gossip/gossip.go:332  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43517" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.657372 105762 gossip/gossip.go:1292  [n3] no incoming or outgoing connections
I180326 15:06:52.660404 106019 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:46029
I180326 15:06:52.690907 105762 gossip/gossip.go:332  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:35701" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.751536 105762 gossip/gossip.go:1292  [n4] no incoming or outgoing connections
I180326 15:06:52.759873 106122 gossip/client.go:129  [n4] started gossip client to 127.0.0.1:46029
I180326 15:06:52.798101 105762 gossip/gossip.go:332  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:34147" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.846139 105762 gossip/gossip.go:1292  [n5] no incoming or outgoing connections
I180326 15:06:52.849407 106278 gossip/client.go:129  [n5] started gossip client to 127.0.0.1:46029
I180326 15:06:52.853020 106261 gossip/server.go:269  [n1] refusing gossip from node 5 (max 3 conns); forwarding to 3 ({tcp 127.0.0.1:35701})
I180326 15:06:52.860195 106278 gossip/client.go:134  [n5] closing client to node 1 (127.0.0.1:46029): received forward from node 1 to 3 (127.0.0.1:35701)
I180326 15:06:52.861508 106277 gossip/gossip.go:1306  [n5] node has connected to cluster via gossip
I180326 15:06:52.864358 106262 gossip/client.go:129  [n5] started gossip client to 127.0.0.1:35701
I180326 15:06:52.911480 105762 gossip/gossip.go:332  [n5] NodeDescriptor set to node_id:5 address:<network_field:"tcp" address_field:"127.0.0.1:35453" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
==================
Read at 0x00c421ff6860 by goroutine 679:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:834 +0x7a
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLivenessAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:781 +0x6e0
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:706 +0xfd
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).heartbeatInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:547 +0x5a7
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:433 +0x37c
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:444 +0x348
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0x14a

Previous write at 0x00c421ff6860 by goroutine 586:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:694 +0x28c4
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:300 +0xb5d
  github.com/cockroachdb/cockroach/pkg/storage_test.runSetupSplitSnapshotRace()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1148 +0x21b
  github.com/cockroachdb/cockroach/pkg/storage_test.TestSplitSnapshotRace_SplitWins()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1259 +0xaf
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Goroutine 679 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0xba
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:411 +0x272
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:833 +0x1adb
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:300 +0xb5d
  github.com/cockroachdb/cockroach/pkg/storage_test.runSetupSplitSnapshotRace()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1148 +0x21b
  github.com/cockroachdb/cockroach/pkg/storage_test.TestSplitSnapshotRace_SplitWins()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1259 +0xaf
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Goroutine 586 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:978 +0x2cd
  github.com/cockroachdb/cockroach/pkg/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x2a8
  main.main()
      _testmain.go:1044 +0x22a
==================
--- FAIL: test/TestSplitSnapshotRace_SplitWins (1.200s)

------- Stdout: -------
I180326 14:51:31.302725 94113 storage/store.go:1394  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180326 14:51:31.302800 94113 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35641" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 14:51:31.308210 94113 gossip/gossip.go:1292  [n2] no incoming or outgoing connections
I180326 14:51:31.308434 94113 storage/store.go:1394  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I180326 14:51:31.308460 94183 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:35641
I180326 14:51:31.308526 94113 gossip/gossip.go:332  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:45061" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 14:51:31.357458 94113 gossip/gossip.go:1292  [n3] no incoming or outgoing connections
I180326 14:51:31.357679 94447 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:35641
I180326 14:51:31.357735 94113 storage/store.go:1394  [s3] [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I180326 14:51:31.357810 94113 gossip/gossip.go:332  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:43565" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 14:51:31.362494 94113 gossip/gossip.go:1292  [n4] no incoming or outgoing connections
I180326 14:51:31.362711 94547 gossip/client.go:129  [n4] started gossip client to 127.0.0.1:35641
I180326 14:51:31.362732 94113 storage/store.go:1394  [s4] [n4,s4]: failed initial metrics computation: [n4,s4]: system config not yet available
I180326 14:51:31.362791 94113 gossip/gossip.go:332  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:37279" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 14:51:31.367266 94113 gossip/gossip.go:1292  [n5] no incoming or outgoing connections
I180326 14:51:31.367454 94596 gossip/client.go:129  [n5] started gossip client to 127.0.0.1:35641
I180326 14:51:31.367517 94113 storage/store.go:1394  [s5] [n5,s5]: failed initial metrics computation: [n5,s5]: system config not yet available
I180326 14:51:31.367620 94113 gossip/gossip.go:332  [n5] NodeDescriptor set to node_id:5 address:<network_field:"tcp" address_field:"127.0.0.1:35727" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I180326 14:51:31.367790 94725 gossip/server.go:269  [n1] refusing gossip from node 5 (max 3 conns); forwarding to 3 ({tcp 127.0.0.1:43565})
I180326 14:51:31.368003 94596 gossip/client.go:134  [n5] closing client to node 1 (127.0.0.1:35641): received forward from node 1 to 3 (127.0.0.1:43565)
I180326 14:51:31.368081 94716 gossip/gossip.go:1306  [n5] node has connected to cluster via gossip
I180326 14:51:31.368176 94751 gossip/client.go:129  [n5] started gossip client to 127.0.0.1:43565
W180326 14:51:31.372306 94113 gossip/gossip.go:1292  [n6] no incoming or outgoing connections
I180326 14:51:31.372508 94113 storage/store.go:1394  [s6] [n6,s6]: failed initial metrics computation: [n6,s6]: system config not yet available
I180326 14:51:31.372509 94849 gossip/client.go:129  [n6] started gossip client to 127.0.0.1:35641
I180326 14:51:31.372566 94113 gossip/gossip.go:332  [n6] NodeDescriptor set to node_id:6 address:<network_field:"tcp" address_field:"127.0.0.1:36385" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I180326 14:51:31.372794 94870 gossip/server.go:269  [n1] refusing gossip from node 6 (max 3 conns); forwarding to 2 ({tcp 127.0.0.1:45061})
I180326 14:51:31.373035 94849 gossip/client.go:134  [n6] closing client to node 1 (127.0.0.1:35641): received forward from node 1 to 2 (127.0.0.1:45061)
I180326 14:51:31.373075 94888 gossip/gossip.go:1306  [n6] node has connected to cluster via gossip
I180326 14:51:31.373204 94191 gossip/client.go:129  [n6] started gossip client to 127.0.0.1:45061
I180326 14:51:31.375079 94113 storage/replica_command.go:862  [s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/Max [r2]
I180326 14:51:31.377987 95060 storage/replica_raftstorage.go:520  [s1,r2/1:/{System/Max-Max}] generated preemptive snapshot 0b509d34 at index 11
I180326 14:51:31.378633 95060 storage/store.go:3828  [s1,r2/1:/{System/Max-Max}] streamed snapshot to (n2,s2):?: kv pairs: 35, log entries: 1, rate-limit: 2.0 MiB/sec, 1ms
I180326 14:51:31.378996 94875 storage/replica_raftstorage.go:726  [s2,r2/?:{-}] applying preemptive snapshot at index 11 (id=0b509d34, encoded size=5437, 1 rocksdb batches, 1 log entries)
I180326 14:51:31.380117 94875 storage/replica_raftstorage.go:732  [s2,r2/?:/{System/Max-Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180326 14:51:31.380729 95060 storage/replica_command.go:1784  [s1,r2/1:/{System/Max-Max}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r2:/{System/Max-Max} [(n1,s1):1, next=2]
I180326 14:51:31.382490 95066 storage/replica.go:3277  [txn=3939bc7f,s1,r2/1:/{System/Max-Max}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
W180326 14:51:31.382924 95066 storage/replica.go:2844  [txn=3939bc7f,s1,r2/1:/{System/Max-Max}] intents not processed as async resolution is disabled
I180326 14:51:31.383335 95068 storage/replica_raftstorage.go:520  [s1,r2/1:/{System/Max-Max}] generated preemptive snapshot 177bc01c at index 13
I180326 14:51:31.384099 95068 storage/store.go:3828  [s1,r2/1:/{System/Max-Max}] streamed snapshot to (n3,s3):?: kv pairs: 37, log entries: 3, rate-limit: 2.0 MiB/sec, 1ms
I180326 14:51:31.384413 95070 storage/replica_raftstorage.go:726  [s3,r2/?:{-}] applying preemptive snapshot at index 13 (id=177bc01c, encoded size=6313, 1 rocksdb batches, 3 log entries)
I180326 14:51:31.385637 95070 storage/replica_raftstorage.go:732  [s3,r2/?:/{System/Max-Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180326 14:51:31.386127 95068 storage/replica_command.go:1784  [s1,r2/1:/{System/Max-Max}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r2:/{System/Max-Max} [(n1,s1):1, (n2,s2):2, next=3]
I180326 14:51:31.387373 95051 storage/raft_transport.go:459  raft transport stream to node 1 established
I180326 14:51:31.390910 95055 storage/replica.go:3277  [txn=6ab0a461,s1,r2/1:/{System/Max-Max}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
W180326 14:51:31.391678 95055 storage/replica.go:2844  [txn=6ab0a461,s1,r2/1:/{System/Max-Max}] intents not processed as async resolution is disabled
I180326 14:51:31.392112 95057 storage/replica_raftstorage.go:520  [s1,r2/1:/{System/Max-Max}] generated preemptive snapshot 1db14a76 at index 16
I180326 14:51:31.392842 95057 storage/store.go:3828  [s1,r2/1:/{System/Max-Max}] streamed snapshot to (n4,s4):?: kv pairs: 39, log entries: 6, rate-limit: 2.0 MiB/sec, 1ms
I180326 14:51:31.393082 94859 storage/replica_raftstorage.go:726  [s4,r2/?:{-}] applying preemptive snapshot at index 16 (id=1db14a76, encoded size=7278, 1 rocksdb batches, 6 log entries)
I180326 14:51:31.394409 94859 storage/replica_raftstorage.go:732  [s4,r2/?:/{System/Max-Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180326 14:51:31.394882 95057 storage/replica_command.go:1784  [s1,r2/1:/{System/Max-Max}] change replicas (ADD_REPLICA (n4,s4):4): read existing descriptor r2:/{System/Max-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4]
I180326 14:51:31.398207 95090 storage/replica.go:3277  [txn=f225923e,s1,r2/1:/{System/Max-Max}] proposing ADD_REPLICA((n4,s4):4): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3 (n4,s4):4] next=5
W180326 14:51:31.399122 95090 storage/replica.go:2844  [txn=f225923e,s1,r2/1:/{System/Max-Max}] intents not processed as async resolution is disabled
I180326 14:51:31.470472 94384 storage/replica_proposal.go:201  [s2,r2/2:/{System/Max-Max}] new range lease repl=(n2,s2):2 seq=2 start=0.000000123,384 epo=1 pro=0.000000123,385 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,6 pro=0.000000123,8
W180326 14:51:31.516498 95083 storage/replica.go:2623  [range-lookup=/System/Max,s1,r1/1:/{Min-System/Max}] intents not processed as async resolution is disabled
I180326 14:51:31.517158 95085 storage/replica_command.go:1784  [s2,r2/2:/{System/Max-Max}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r2:/{System/Max-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n4,s4):4, next=5]
I180326 14:51:31.521005 94935 storage/replica.go:3277  [txn=4305ca36,s2,r2/2:/{System/Max-Max}] proposing REMOVE_REPLICA((n1,s1):1): updated=[(n4,s4):4 (n2,s2):2 (n3,s3):3] next=5
W180326 14:51:31.521811 94935 storage/replica.go:2844  [txn=4305ca36,s2,r2/2:/{System/Max-Max}] intents not processed as async resolution is disabled
I180326 14:51:31.522183 94990 storage/store.go:3543  [s1,r2/1:/{System/Max-Max}] added to replica GC queue (peer suggestion)
I180326 14:51:31.523443 95093 storage/store.go:2414  [replicaGC,s1,r2/1:/{System/Max-Max}] removing replica
I180326 14:51:31.523716 95093 storage/replica.go:836  [replicaGC,s1,r2/1:/{System/Max-Max}] removed 39 (27+12) keys in 0ms [clear=0ms commit=0ms]
I180326 14:51:31.524058 94113 storage/client_test.go:1221  test clock advanced to: 1.800000125,0
I180326 14:51:31.526093 94394 storage/replica_proposal.go:201  [s2,r2/2:/{System/Max-Max}] new range lease repl=(n2,s2):2 seq=2 start=0.000000123,384 epo=1 pro=1.800000125,5 following repl=(n2,s2):2 seq=2 start=0.000000123,384 epo=1 pro=0.000000123,385
I180326 14:51:31.526782 95019 storage/replica_command.go:862  [s2,r2/2:/{System/Max-Max}] initiating a split of this range at key "m" [r11]
I180326 14:51:31.530984 94397 storage/replica_proposal.go:201  [txn=37885193,s2,r2/2:/{System/Max-Max}] new range lease repl=(n2,s2):2 seq=2 start=0.000000123,384 epo=1 pro=1.800000125,5 following repl=(n2,s2):2 seq=2 start=0.000000123,384 epo=1 pro=1.800000125,5
W180326 14:51:31.531291 95139 storage/replica.go:2844  [txn=37885193,s2,r2/2:{/System/Max-m}] intents not processed as async resolution is disabled
I180326 14:51:31.532010 95142 storage/replica_raftstorage.go:520  [s2,r11/2:{m-/Max}] generated preemptive snapshot e5995d18 at index 10
I180326 14:51:31.532741 95142 storage/store.go:3828  [s2,r11/2:{m-/Max}] streamed snapshot to (n5,s5):?: kv pairs: 34, log entries: 0, rate-limit: 2.0 MiB/sec, 1ms
I180326 14:51:31.533032 95159 storage/replica_raftstorage.go:726  [s5,r11/?:{-}] applying preemptive snapshot at index 10 (id=e5995d18, encoded size=5414, 1 rocksdb batches, 0 log entries)
I180326 14:51:31.533778 95159 storage/replica_raftstorage.go:732  [s5,r11/?:{m-/Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180326 14:51:31.534445 95142 storage/replica_command.go:1784  [s2,r11/2:{m-/Max}] change replicas (ADD_REPLICA (n5,s5):4): read existing descriptor r11:{m-/Max} [(n4,s4):1, (n2,s2):2, (n3,s3):3, next=4]
I180326 14:51:31.539995 95171 storage/replica.go:3277  [txn=072a5f13,s2,r11/2:{m-/Max}] proposing ADD_REPLICA((n5,s5):4): updated=[(n4,s4):1 (n2,s2):2 (n3,s3):3 (n5,s5):4] next=5
W180326 14:51:31.540870 95171 storage/replica.go:2844  [txn=072a5f13,s2,r11/2:{m-/Max}] intents not processed as async resolution is disabled
I180326 14:51:31.541542 95127 storage/replica_raftstorage.go:520  [s2,r11/2:{m-/Max}] generated preemptive snapshot e36ec665 at index 14
I180326 14:51:31.542161 95127 storage/store.go:3828  [s2,r11/2:{m-/Max}] streamed snapshot to (n6,s6):?: kv pairs: 36, log entries: 4, rate-limit: 2.0 MiB/sec, 1ms
I180326 14:51:31.542388 95148 storage/replica_raftstorage.go:726  [s6,r11/?:{-}] applying preemptive snapshot at index 14 (id=e36ec665, encoded size=6837, 1 rocksdb batches, 4 log entries)
I180326 14:51:31.543612 95148 storage/replica_raftstorage.go:732  [s6,r11/?:{m-/Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180326 14:51:31.544082 95127 storage/replica_command.go:1784  [s2,r11/2:{m-/Max}] change replicas (ADD_REPLICA (n6,s6):5): read existing descriptor r11:{m-/Max} [(n4,s4):1, (n2,s2):2, (n3,s3):3, (n5,s5):4, next=5]
I180326 14:51:31.548263 94942 storage/replica.go:3277  [txn=160442f2,s2,r11/2:{m-/Max}] proposing ADD_REPLICA((n6,s6):5): updated=[(n4,s4):1 (n2,s2):2 (n3,s3):3 (n5,s5):4 (n6,s6):5] next=6
W180326 14:51:31.548958 94942 storage/replica.go:2844  [txn=160442f2,s2,r11/2:{m-/Max}] intents not processed as async resolution is disabled
I180326 14:51:31.559276 95053 storage/store.go:3562  [s2,r2/2:{/System/Max-m}] added to replica GC queue (contacted deleted peer)
I180326 14:51:31.568124 95162 storage/replica_command.go:1784  [s2,r11/2:{m-/Max}] change replicas (REMOVE_REPLICA (n3,s3):3): read existing descriptor r11:{m-/Max} [(n4,s4):1, (n2,s2):2, (n3,s3):3, (n5,s5):4, (n6,s6):5, next=6]
I180326 14:51:31.570162 95186 storage/replica.go:3277  [txn=b7b96059,s2,r11/2:{m-/Max}] proposing REMOVE_REPLICA((n3,s3):3): updated=[(n4,s4):1 (n2,s2):2 (n6,s6):5 (n5,s5):4] next=6
W180326 14:51:31.571073 95186 storage/replica.go:2844  [txn=b7b96059,s2,r11/2:{m-/Max}] intents not processed as async resolution is disabled
I180326 14:51:31.571313 94990 storage/store.go:3543  [s3,r11/3:{m-/Max}] added to replica GC queue (peer suggestion)
I180326 14:51:31.572892 94791 storage/replica_proposal.go:201  [s5,r11/4:{m-/Max}] new range lease repl=(n5,s5):4 seq=3 start=1.800000125,283 epo=1 pro=1.800000125,284 following repl=(n2,s2):2 seq=2 start=0.000000123,384 epo=1 pro=1.800000125,5
I180326 14:51:31.573332 95165 storage/store.go:2414  [replicaGC,s3,r11/3:{m-/Max}] removing replica
I180326 14:51:31.573640 95165 storage/replica.go:836  [replicaGC,s3,r11/3:{m-/Max}] removed 37 (26+11) keys in 0ms [clear=0ms commit=0ms]
I180326 14:51:31.621128 95226 storage/replica_command.go:1784  [s5,r11/4:{m-/Max}] change replicas (REMOVE_REPLICA (n2,s2):2): read existing descriptor r11:{m-/Max} [(n4,s4):1, (n2,s2):2, (n6,s6):5, (n5,s5):4, next=6]
I180326 14:51:31.623387 95032 storage/replica.go:3277  [txn=e861eb75,s5,r11/4:{m-/Max}] proposing REMOVE_REPLICA((n2,s2):2): updated=[(n4,s4):1 (n5,s5):4 (n6,s6):5] next=6
W180326 14:51:31.624301 95032 storage/replica.go:2844  [txn=e861eb75,s5,r11/4:{m-/Max}] intents not processed as async resolution is disabled
I180326 14:51:31.624779 95130 storage/store.go:3543  [s2,r11/2:{m-/Max}] added to replica GC queue (peer suggestion)
I180326 14:51:31.625888 95189 storage/store.go:2414  [replicaGC,s2,r11/2:{m-/Max}] removing replica
I180326 14:51:31.626208 95189 storage/replica.go:836  [replicaGC,s2,r11/2:{m-/Max}] removed 38 (26+12) keys in 0ms [clear=0ms commit=0ms]
I180326 14:51:32.131076 95117 storage/client_test.go:1221  test clock advanced to: 3.600000127,0
I180326 14:51:32.132277 95447 storage/replica_proposal.go:201  [s4,r2/4:{/System/Max-m}] new range lease repl=(n4,s4):4 seq=3 start=1.800000125,600 epo=1 pro=1.800000125,601 following repl=(n2,s2):2 seq=2 start=0.000000123,384 epo=1 pro=1.800000125,5
I180326 14:51:32.134833 95467 storage/replica_proposal.go:201  [s4,r2/4:{/System/Max-m}] new range lease repl=(n4,s4):4 seq=4 start=1.800000125,600 epo=2 pro=3.600000127,19 following repl=(n4,s4):4 seq=3 start=1.800000125,600 epo=1 pro=1.800000125,601
I180326 14:51:32.442998 95731 storage/replica_proposal.go:201  [s6,r11/5:{m-/Max}] new range lease repl=(n6,s6):5 seq=4 start=3.600000127,159 epo=1 pro=3.600000127,160 following repl=(n5,s5):4 seq=3 start=1.800000125,283 epo=1 pro=1.800000125,284
W180326 14:51:32.444090 95675 storage/store.go:1486  [s5,r11/4:{m-/Max}] could not gossip system config: [NotLeaseHolderError] r11: replica (n5,s5):4 not lease holder; lease holder unknown
I180326 14:51:32.444168 95827 storage/client_test.go:1221  test clock advanced to: 5.400000129,0
I180326 14:51:32.444953 95744 storage/replica_proposal.go:201  [s6,r11/5:{m-/Max}] new range lease repl=(n6,s6):5 seq=5 start=3.600000127,159 epo=2 pro=3.600000127,269 following repl=(n6,s6):5 seq=4 start=3.600000127,159 epo=1 pro=3.600000127,160
I180326 14:51:32.445466 95830 storage/client_test.go:1221  test clock advanced to: 7.200000131,0
I180326 14:51:32.447621 95746 storage/replica_proposal.go:201  [s6,r11/5:{m-/Max}] new range lease repl=(n6,s6):5 seq=5 start=3.600000127,159 epo=2 pro=5.400000129,25 following repl=(n6,s6):5 seq=5 start=3.600000127,159 epo=2 pro=3.600000127,269
I180326 14:51:32.449065 95741 storage/replica_proposal.go:201  [s6,r11/5:{m-/Max}] new range lease repl=(n6,s6):5 seq=5 start=3.600000127,159 epo=2 pro=7.200000131,38 following repl=(n6,s6):5 seq=5 start=3.600000127,159 epo=2 pro=5.400000129,25
W180326 14:51:32.493747 95051 storage/raft_transport.go:465  raft transport stream to node 1 failed: EOF
W180326 14:51:32.493816 94944 storage/raft_transport.go:465  raft transport stream to node 6 failed: EOFRace detected!

------- Stdout: -------
I180326 15:06:52.432640 105762 storage/store.go:1394  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180326 15:06:52.471270 105762 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46029" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.527940 105762 gossip/gossip.go:1292  [n2] no incoming or outgoing connections
I180326 15:06:52.538151 105883 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:46029
I180326 15:06:52.553567 105762 storage/store.go:1394  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I180326 15:06:52.559421 105762 gossip/gossip.go:332  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43517" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.657372 105762 gossip/gossip.go:1292  [n3] no incoming or outgoing connections
I180326 15:06:52.660404 106019 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:46029
I180326 15:06:52.690907 105762 gossip/gossip.go:332  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:35701" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.751536 105762 gossip/gossip.go:1292  [n4] no incoming or outgoing connections
I180326 15:06:52.759873 106122 gossip/client.go:129  [n4] started gossip client to 127.0.0.1:46029
I180326 15:06:52.798101 105762 gossip/gossip.go:332  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:34147" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180326 15:06:52.846139 105762 gossip/gossip.go:1292  [n5] no incoming or outgoing connections
I180326 15:06:52.849407 106278 gossip/client.go:129  [n5] started gossip client to 127.0.0.1:46029
I180326 15:06:52.853020 106261 gossip/server.go:269  [n1] refusing gossip from node 5 (max 3 conns); forwarding to 3 ({tcp 127.0.0.1:35701})
I180326 15:06:52.860195 106278 gossip/client.go:134  [n5] closing client to node 1 (127.0.0.1:46029): received forward from node 1 to 3 (127.0.0.1:35701)
I180326 15:06:52.861508 106277 gossip/gossip.go:1306  [n5] node has connected to cluster via gossip
I180326 15:06:52.864358 106262 gossip/client.go:129  [n5] started gossip client to 127.0.0.1:35701
I180326 15:06:52.911480 105762 gossip/gossip.go:332  [n5] NodeDescriptor set to node_id:5 address:<network_field:"tcp" address_field:"127.0.0.1:35453" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
==================
Read at 0x00c421ff6860 by goroutine 679:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:834 +0x7a
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLivenessAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:781 +0x6e0
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:706 +0xfd
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).heartbeatInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:547 +0x5a7
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:433 +0x37c
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:444 +0x348
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0x14a

Previous write at 0x00c421ff6860 by goroutine 586:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:694 +0x28c4
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:300 +0xb5d
  github.com/cockroachdb/cockroach/pkg/storage_test.runSetupSplitSnapshotRace()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1148 +0x21b
  github.com/cockroachdb/cockroach/pkg/storage_test.TestSplitSnapshotRace_SplitWins()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1259 +0xaf
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Goroutine 679 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0xba
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:411 +0x272
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:833 +0x1adb
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:300 +0xb5d
  github.com/cockroachdb/cockroach/pkg/storage_test.runSetupSplitSnapshotRace()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1148 +0x21b
  github.com/cockroachdb/cockroach/pkg/storage_test.TestSplitSnapshotRace_SplitWins()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1259 +0xaf
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Goroutine 586 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:978 +0x2cd
  github.com/cockroachdb/cockroach/pkg/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x2a8
  main.main()
      _testmain.go:1044 +0x22a
==================

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -831,7 +831,7 @@ func (m *multiTestContext) addStore(idx int) {
 		ch: make(chan struct{}),
 	}
 	m.nodeLivenesses[idx].StartHeartbeat(ctx, stopper, func(ctx context.Context) {
-		now := m.clocks[idx].Now()
+		now := clock.Now()
 		if err := store.WriteLastUpTimestamp(ctx, now); err != nil {
 			log.Warning(ctx, err)
 		}

